edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/modules/CommonModule.cs

```diff
@@ -2,35 +2,235 @@
+                .As<ISystemEnvironment>()
+                .SingleInstance();
+
+            // DataBase options
+            builder.Register(c => new Storage.RocksDb.RocksDbOptionsProvider(c.Resolve<ISystemEnvironment>(), this.optimizeForPerformance))
+                .As<Storage.RocksDb.IRocksDbOptionsProvider>()
+                .SingleInstance();
+
+            // IDbStoreProvider
+            builder.Register(
+                c =>
+                {
+                    var loggerFactory = c.Resolve<ILoggerFactory>();
+                    ILogger logger = loggerFactory.CreateLogger(typeof(RoutingModule));
+
+                    if (this.usePersistentStorage)
+                    {
+                        // Create partitions for messages and twins
+                        var partitionsList = new List<string> { Core.Constants.MessageStorePartitionKey, Core.Constants.TwinStorePartitionKey, Core.Constants.CheckpointStorePartitionKey };
+                        try
+                        {
+                            IDbStoreProvider dbStoreprovider = Storage.RocksDb.DbStoreProvider.Create(c.Resolve<Storage.RocksDb.IRocksDbOptionsProvider>(),
+                                this.storagePath, partitionsList);
+                            logger.LogInformation($"Created persistent store at {this.storagePath}");
+                            return dbStoreprovider;
+                        }
+                        catch (Exception ex) when (!ExceptionEx.IsFatal(ex))
+                        {
+                            logger.LogError(ex, "Error creating RocksDB store. Falling back to in-memory store.");
+                            return new InMemoryDbStoreProvider();
+                        }
+                    }
+                    else
+                    {
+                        logger.LogInformation($"Using in-memory store");
+                        return new InMemoryDbStoreProvider();
+                    }
+                })
+                .As<IDbStoreProvider>()
+                .SingleInstance();
+
+            // Task<IEncryptionProvider>
+            builder.Register(
+                    async c =>
+                    {
+                        IEncryptionProvider encryptionProvider = await this.workloadUri.Map(
+                                async uri => await EncryptionProvider.CreateAsync(
+                                    this.storagePath,
+                                    new Uri(uri),
+                                    Service.Constants.WorkloadApiVersion,
+                                    this.edgeHubModuleId,
+                                    this.edgeHubGenerationId.Expect(() => new InvalidOperationException("Missing generation ID")),
+                                    Service.Constants.InitializationVectorFileName) as IEncryptionProvider)
+                            .GetOrElse(() => Task.FromResult<IEncryptionProvider>(NullEncryptionProvider.Instance));
```

Comment:
Should we log something when we have to fall back to the null encryption provider?

Reply:
I would love to, but I don't think a logger is available at this point. We might be able to get it to work somehow, but in that case maybe i can do it in the next PR.

Reply:
No worries.